src/goto-instrument/nondet_static.cpp

```diff
@@ -50,7 +50,9 @@ void nondet_static(
         continue;
 
       // constant?
-      if(sym.type().get_bool(ID_C_constant))
+      if(sym.type().get_bool(ID_C_constant)
+        // if it's an array, we need to check the subtype
+        || sym.type().subtype().get_bool(ID_C_constant))
```

Comment:
Should you include a test for ID_array as part of the new condition? I'd feel happier if you explicitly tested for (pseudo code) `sym is array && sym subtype is constant`

Reply:
Also this will require recursing to avoid the same problem for multi-dimensional arrays.